### PR TITLE
Fix gridspec.Gridspec: check ratios for consistency with rows and columns

### PR DESCRIPTION
**Pull request by julianvmodesto**
#5835

### CODE REVIEW

File: lib/matplotlib/tests/test_gridspec.py [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
     assert_equal(gs[0, 0], gs[0, 0])
     assert_equal(gs[:, 0], gs[:, 0])
+
+
+def test_ratios():
+    """
+    Addresses issue #5835.
+    See at https://github.com/matplotlib/matplotlib/issues/5835.
+    """
+    assert_raises(ValueError, gridspec.GridSpec,
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_gridspec.py]: Can you add a second one of these which fails via the other path (if I am reading this right this only really tests the width based failure).

**Comment by tacaswell** [comment]
Good to see other new yorkers around :smile:

**Comment by julianvmodesto** [comment]
Hey there! :smile: :statue_of_liberty: I'll add one. I might reopen a new pull request since I put this on my master branch, and maybe I could fork from the 2.x branch to match the milestone.

**Comment by tacaswell** [comment]
Don't worry about that, we are pretty good at back porting

**Comment by jenshnielsen** [comment]
Backported to 2.x as 308ce5d49f455c9698de75f7e93d12c8a312f2e0

**Comment by jenshnielsen** [comment]
Thanks for the fix much appreciated.

**Pull request merged**